### PR TITLE
Fix `new` usage for builtin objects

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
question: keep the rule?

File: scripts/build/build.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ function fitTerminal(input) {
   const columns = Math.min(process.stdout.columns || 40, 80);
   const WIDTH = columns - stringWidth(OK) + 1;
   if (input.length < WIDTH) {
-    input += Array(WIDTH - input.length).join(chalk.dim("."));
+    input += new Array(WIDTH - input.length).join(chalk.dim("."));
```
Comment (lydell) [inline scripts/build/build.js]: We could also change stuff like this into `.repeat` on the `next` branch.

**Comment by lydell** [comment]
Maybe it’s going to be easier to merge all the eslint-plugin-unicorn PRs into `next`?

**Comment by evilebottnawi** [comment]
@lydell why? It is not breaking change :smile:

**Comment by lydell** [comment]
I thought about merge conflicts. If we merge this to master and then replace `Array(n+1).join("x")` with `"x".repeat(n)` in `next` we will get a lot of unnecessary conflicts.

**Comment by evilebottnawi** [comment]
@lydell it makes sense :+1:

File: scripts/build/build.js (hunk 1 above)
Comment (lydell) [reply scripts/build/build.js]: /cc @fisker if you’re looking for more refactoring to do :)

Comment (fisker) [reply scripts/build/build.js]: maybe later

**Pull request merged**